test/common/thread_local/thread_local_impl_test.cc

```diff
@@ -82,6 +82,32 @@ TEST_F(ThreadLocalInstanceImplTest, All) {
   tls_.shutdownThread();
 }
 
+// Validate ThreadLocal::runOnAllThreads behavior with all_thread_complete call back.
+TEST_F(ThreadLocalInstanceImplTest, RunOnAllThreads) {
+  SlotPtr tlsptr = tls_.allocateSlot();
+
+  EXPECT_CALL(thread_dispatcher_, post(_));
+  EXPECT_CALL(main_dispatcher_, post(_));
+
+  // Ensure that the thread local call back and all_thread_complete call back are called.
+  struct {
+    uint64_t thread_local_calls_{0};
+    bool all_threads_complete_ = false;
+  } thread_local_data;
```

Comment:
I don't think this data is thread-local.  Maybe just rename the struct to be 'thread_status'